resin/models/config_variable.py

```diff
@@ -0,0 +1,262 @@
+import re
+import json
+
+from ..base_request import BaseRequest
+from .device import Device
+from ..settings import Settings
+from .. import exceptions
+
+
+def _is_valid_config_var_name(config_var_name):
+    return config_var_name.startswith('RESIN_')
+
+
+class ConfigVariable(object):
+    """
+    This class is a wrapper for config variable models.
+
+    """
+
+    def __init__(self):
+        self.device_config_variable = DeviceConfigVariable()
+        self.application_config_variable = ApplicationConfigVariable()
+
+
+class DeviceConfigVariable(object):
+    """
+    This class implements device config variable model for Resin Python SDK.
+
+    """
+
+    def __init__(self):
+        self.base_request = BaseRequest()
+        self.device = Device()
+        self.settings = Settings()
+
+    def get_all(self, uuid):
+        """
+        Get all device config variables belong to a device.
+
+        Args:
+            uuid (str): device uuid.
+
+        Returns:
+            list: device config variables.
+
+        Examples:
+            >>> resin.models.config_variable.device_config_variable.get_all('f5213eac0d63ac47721b037a7406d306')
+            [{u'device': {u'__deferred': {u'uri': u'/resin/device(1036574)'}, u'__id74}, u'__metadata': {u'type': u'', u'uri': u'/resin/device_config_variab8)'}, u'id': 130598, u'value': u'1', u'name': u'RESIN_HOST_CONFIG_avoid_'}, {u'device': {u'__deferred': {u'uri': u'/resin/device(1036574)'}, u'_36574}, u'__metadata': {u'type': u'', u'uri': u'/resin/device_config_var0597)'}, u'id': 130597, u'value': u'1', u'name': u'RESIN_HOST_CONFIG_disash'}, {u'device': {u'__deferred': {u'uri': u'/resin/device(1036574)'},  1036574}, u'__metadata': {u'type': u'', u'uri': u'/resin/device_config_(130596)'}, u'id': 130596, u'value': u'"i2c_arm=on","spi=on","audio=on"'': u'RESIN_HOST_CONFIG_dtparam'}, {u'device': {u'__deferred': {u'uri': udevice(1036574)'}, u'__id': 1036574}, u'__metadata': {u'type': u'', u'uresin/device_config_variable(130595)'}, u'id': 130595, u'value': u'16', uu'RESIN_HOST_CONFIG_gpu_mem'}, {u'device': {u'__deferred': {u'uri': u'/rice(1036574)'}, u'__id': 1036574}, u'__metadata': {u'type': u'', u'uri':n/device_config_variable(130594)'}, u'id': 130594, u'value': u'false', uu'RESIN_HOST_LOG_TO_DISPLAY'}]
+
+        """
+
+        device = self.device.get(uuid)
+
+        params = {
+            'filter': 'device',
+            'eq': device['id']
+        }
+
+        return self.base_request.request(
+            'device_config_variable', 'GET', params=params,
+            endpoint=self.settings.get('pine_endpoint')
+        )['d']
```

Comment:
This does two requests to get the vars - that works, but it is possible to do a pine query that gets all device_config_variables from a UUID directly, in just one request. Not sure how concerned we are about the performance implications here, but might be worth looking at (same applies for many of the below too)